src/pages/AddItem.js

```diff
@@ -14,24 +34,62 @@ export default function AddItem(props) {
     setPurchaseFrequency(e.target.value);
   };
 
-  const createListItem = (e) => {
-    e.preventDefault();
-    db.collection('generated_token')
-      .add({
-        item_name: itemName,
-        purchase_frequency: parseInt(purchaseFrequency),
-        last_purchased: lastPurchased,
-      })
-      .then((documentReference) => {
-        console.log('document reference ID', documentReference.id);
-      })
-      .catch((error) => {
-        console.log(error.message);
-      });
-    setItemName('');
-    setLastPurchased(null);
+  const checkForDuplicates = () => {
+    // we would like to potentially normalize white space but unsuccessfull so far
+    const normalizedUserInput = itemName
+      .toLowerCase()
+      .replace(/[^\w\s]|_/g, '');
+
+    const matchingItemName = listItems.filter((item) => {
+      const normalizedDataBaseItem = item.item_name
+        .toLowerCase()
+        .replace(/[^\w\s]|_/g, '');
+      console.log(normalizedDataBaseItem);
+      return normalizedDataBaseItem === normalizedUserInput;
+    });
+
+    if (matchingItemName.length > 0) {
+      return false;
+    } else {
+      return true;
+    }
```

Comment:
maybe instead of an `if` statement we could just shorten this using `return !matchingItemName` instead, since we want to flip the truthy or falsy value of our condition.

Reply:
@apdsrocha Yeah! I was talking to Jessica about how I know there's probably a better way, but I just couldn't remember what it was! Thanks for the suggestion!

Reply:
`return !matchingItemName` doesn't achieve the desired result. We were able to change the logic in the if statement to `if (matchingItemName === 0)` and `return true` which allows the item to hit the database. And if `matchingItemName` is greater than `0` it returns `undefined` which is a falsey value, so it works, but it doesn't seem like best practices.

Reply:
@jamesncox What about `return !matchingItemName.length>0`

Reply:
@laurenyz I did try your suggestion and was unable to get it to work as intended. Instead, I would receive an error message telling me the item was already in the list, even if it was not.

Reply:
@jssckbl that's so strange! I was able to getting working locally - we can add it to the long list of things to talk about tonight 😂

Reply:
Today (4/22) Jessica and I refactored a lot of things we talked about last night. We renamed `checkForDuplicates()` to `doesItemExistInDatabase()` and refactored the final return to `return matchingItemName.length === 0;` which gives us the desired result. Basically if the `matchingItemName` length is zero, it will allow the item to hit the database. Otherwise, it will return false, and will throw the error/alert.

Reply:
Quick update on doesItemExistInDatabase() return:

I changed it to `return matchingItemName.length !== 0;` so that it returns `true` if an item does exist and `false` if an item does not already exist in Firestore.

This allowed me to update `createListItem()` to make the function more readable:

```javascript
function createListItem(e) {
    e.preventDefault();

    const newItemObject = {
      item_name: itemName,
      purchase_frequency: parseInt(purchaseFrequency),
      last_purchased: lastPurchased,
    };

    const itemExists = doesItemExistInDatabase(itemName);

    if (itemExists) {
      swal(
        'OH GOSH!',
        `${itemName
          .toUpperCase()
          .replace(/[^\w\s]|_/g, '')} is already in your list`,
        'error',
      );
    } else if (!itemName) {
      swal('UH OH!', "Item name can't be blank", 'warning');
    } else {
      db.collection(props.token).add(newItemObject);
      props.setListItems((prev) => [...prev, newItemObject]);
      history.push('/list');
    }
  }
```
Finally, after reviewing the other team's PR, I realized we didn't add a check for empty input: `itemName === ""` was getting through to the data base, so I added 
```javascript 
    } else if (!itemName) {
      swal('UH OH!', "Item name can't be blank", 'warning');
```